### ISSUE TITLE
Feature requests: Commands

### ISSUE THREAD

**Issue by SlimeDog** [broccolai/PureTickets#6]
### /ticket list
The list command needs to be filtered to provide targeted information. Suggestions:
- /ticket list [ --new | --open | --closed | --picked ] [playerName]
- /ticket list [playerName] [ --new | --open | --closed | --picked ]

Details:
- The playerName and filters can be specified in any order.
- Only one filter may be specified? Or if more than one, project the union?
- Tab-completion should apply to filters, so that one needs to type only a few letters.

### /ticket status
- /ticket status [staffName]

Details:
- Produce a summary of all tickets, grouped by status.
- If a staffName is specified, filter only tickets picked/assigned to specified staff.

For example:
```
/ticket status
==== TICKET STATUS ====
27 Open, unpicked
 5 Open, picked
10 Closed
```
```
/ticket status SlimeDog
==== TICKET STATUS FOR SlimeDog ====
 2 Open, picked
 3 Closed
```
where
- Open, unpicked = Open, but unpicked -- Does not apply if staff specified
- Open, picked = Open and picked/assigned
- Closed = Self-evident

**Comment by broccolai** [comment]
Sounds good, I’ll add this today

**Comment by broccolai** [comment]
Couldn't figure out a good way for the flags, for now I've added a ticket status filter to the list command, I'll look more at this tomorrow.

As for the status command it's added

**Comment by SlimeDog** [comment]
```
Couldn't figure out a good way for the flags
```
Why not the double-hyphens for flags?

**Comment by broccolai** [comment]
I meant I couldn't find a good way to implement it with my current command system while retaining command competitions for all of arguments. It's probably just something I'm missing, I'm going to look into it more though

**Comment by SlimeDog** [comment]
Ah, I understand. Thanks.

**Comment by SlimeDog** [comment]
PT 2.1.0 reflections:
- The <player> parameter should be removed from all commands. Tickets are indexed universally, not by player.
- If a filter is employed on a list, the title should reflect the filter; "ALL TICKETS" is misleading.
- All placeholders should be recognized in (nearly) all messages, where a value is known.
- BIG ISSUE: If errors are discovered in config.yml or locale file, errors should be printed and logged, and the plugin disabled if necessary (but that would be a rare event). If the defaults must be used (for example, if the errors are too severe), then print an error message to that effect. But in no case should the files be rewritten with the defaults.

**Comment by SlimeDog** [comment]
Output of /ticket list ... should be
**id player status title**
For example:
```
/ticket list SlimeDog
  id  player    status  staff      title
   1  SlimeDog  open               Help me please
  12  SlimeDog  closed  broccolai  Where is spawn?
  33  SlimeDog  open               How do I change worlds?
 154  SlimeDog  open               Can I go to whitedog?
```
id = index
reporter = player who created it
status = open | picked | closed | reopened
title = creation message

**Comment by broccolai** [comment]
Not really a fan of that design tbh, but I'll make it a locale option so you can customise it. Working on all the others though, I'm hesitant to remove the player parameter because I've found it hard to remember the ticket ids the whole time. At least thats my experience with server moderation.  If I can find a nice way to make that a configurable option I'll add it.

**Comment by RogerAntonio** [comment]
What about 


**/ticket list [open|closed] [player]**
Lists all tickets
Optional lists all open or closed tickets
And on top of that optionally lists all open or closed tickets by playername
And then when you found the ticket
**/ticket show <id>** (no playername)

**Comment by SlimeDog** [comment]
```
I'm hesitant to remove the player parameter because I've found it hard to remember the ticket ids the whole time
```
**/ticket list <player>**
gives you the ids whenever you need them. I don't understand the other way, since the ids are not player-specific. Perhaps I'm missing something.

**Comment by broccolai** [comment]
It's more about the relying on tab competition yeah, if I make it only a take a id you lose the ability to do /ti pick broccolai. Using it without id's is how I use it most of the time because for the most part I'm replying to the most recent tickets. 

I might be able to make it /ticket show <playername or id> [id], but the first id parameter wouldn't have tab completion

**Comment by RogerAntonio** [comment]
```===  TICKET COMMANDS  === 
ticket log <Player> [Index] - Log tickets messages
ticket teleport <Player> [Index] - Teleport to a ticket creation location
ticket show <Player> [Index] - Show a ticket
ticket update <Index> <Message> - Update a ticket
ticket pick <Player> [Index] - Pick a ticket
ticket yield <Player> [Index] - Yield a ticket
ticket create <Message> - Create a ticket
ticket close [Index] - Close a ticket
ticket done <Player> [Index] - Done-mark a ticket
ticket list [Player] - List all tickets
ticket reopen <Player> [Index] - Reopen a ticket
ticket status [Player] - View amount of tickets in
```

All the `<Player>` arguments are redundant as we lookup tickets by ID anyway.
Forgot the ID? Simple, type /ticket list or /ticket list [Player|open|closed] to list the tickets by a player.
Also, remember: `<required> [optional]`

Corrected commands:
```===  TICKET COMMANDS  === 
ticket log <Index> - Log tickets messages
ticket teleport <Index> - Teleport to a ticket creation location
ticket show <Index> - Show a ticket
ticket update <Index> <Message> - Update a ticket
ticket pick  <Index> - Pick a ticket
ticket yield  <Index> - Yield a ticket
ticket create <Message> - Create a ticket
ticket close <Index> - Close a ticket
ticket done <Index> - Done-mark a ticket
ticket list [Player|open|closed] - List all tickets 
ticket reopen <Index> - Reopen a ticket
ticket status [Player] - View amount of tickets in [by playername]
```

**Comment by broccolai** [comment]
Yeah I know, the id's / indexs are optional. You can just use the player name and it'll get the most recent ticket from them. That's why I like it, like I said if I can make it configurable I'll give an option between the two

**Comment by broccolai** [comment]
> All placeholders should be recognized in (nearly) all messages, where a value is known.

This is added in 2.1.2 and you can find the replacements you can use here
https://github.com/broccolai/PureTickets/blob/f5b7fc0b1e34135993b7728ce2b95e24b8db2c23/src/main/kotlin/co/uk/magmo/puretickets/utils/Utils.kt#L19

**Comment by broccolai** [comment]
Closing this now, I believe most requests have been fixed in 2.2. If there are any more requests please let me know on separate issues.

I have decided not to implement the id only commands, they don't really fit with what I'm going with and having them as an option would mean I'd have to support the two different command types every time I make changes, which isn't really something I'd like to do.

If that's a deal breaker to you then I'm really sorry and your input has been crucial in making the plugin better and I'm grateful for it, thank you.

**Comment by SlimeDog** [comment]
Thanks. I appreciate your responsiveness and everything you've done. It's great that you have shared this work.
